backends/rpm: fix parsing of rpm-md metadata

If assertions are disabled then the queue.pop() wasn't being executed,
leading to requires, recommends etc. not being read properly.

Upstream-Status: Pending

Signed-off-by: Paul Eggleton <paul.eggleton@linux.intel.com>

diff --git a/smart/backends/rpm/metadata.py b/smart/backends/rpm/metadata.py
index 2c54f39..dc9df22 100644
--- a/smart/backends/rpm/metadata.py
+++ b/smart/backends/rpm/metadata.py
@@ -188,7 +188,8 @@ class RPMMetaDataLoader(Loader):
 
             elif event == "end":
 
-                assert queue.pop() is elem
+                popped = queue.pop()
+                assert popped is elem
 
                 if skip:
                     if tag == skip:
-- 
1.7.9.5

